### ISSUE TITLE
Installation issue: python 3.7.4 package fails to build with very recent spack checkout

### ISSUE THREAD

**Issue by hppritcha** [spack/spack#14603]
### Steps to reproduce the issue

checkout out spack at sha  5eed196f

```console
$ spack install python@3.7.4  % gcc@9.1.0
```

Gets to building and dies with this error message:

```
==> Error: ProcessError: Command exited with status -6:
    '/projects/artab/users/hpp/spack-environments-cc/DWARMG_9.1.0/1.0/spack/opt/spack/linux-rhel7-aarch64/gcc-9.1.0/python-3.7.4-3op4hzriixljjkkqjjv4bsuk7lc3ripe/bin/python3.7' '-c' 'from distutils.sysconfig import get_config_var; print(get_config_var('"'"'LIBDEST'"'"'))'

1 error found in build log:
     8982    		install|*) ensurepip="" ;; \
     8983    	esac; \
     8984    	LD_LIBRARY_PATH=/ram/tmp/hpp/spack-stage/spack-stage-python-3.7.4-3op4hzriixljjkkqjjv4bsuk7lc3ripe/spack-src/spack-build:/projects/opt
             /aarch64/gcc/9.1.0/lib64:/projects/opt/aarch64/gcc/9.1.0/lib:/usr/local/lib ./python -E -m ensurepip \
     8985    		$ensurepip --root=/ ; \
     8986    fi
     8987    ==> [2020-01-22-14:37:30.087737] '/projects/artab/users/hpp/spack-environments-cc/DWARMG_9.1.0/1.0/spack/opt/spack/linux-rhel7-aarch64/
             gcc-9.1.0/python-3.7.4-3op4hzriixljjkkqjjv4bsuk7lc3ripe/bin/python3.7' '-c' 'from distutils.sysconfig import get_config_var; print(get_
             config_var('"'"'LIBDEST'"'"'))'
  >> 8988    Fatal Python error: initfsencoding: unable to load the file system codec
     8989    ModuleNotFoundError: No module named 'encodings'
     8990    
     8991    Current thread 0x0000400033217750 (most recent call first):

See build log for details:
  /tmp/hpp/spack-stage/spack-stage-python-3.7.4-3op4hzriixljjkkqjjv4bsuk7lc3ripe/spack-build-out.txt

```

### Platform and user environment

Please report your OS here:
```commandline
$ uname -a 
Linux cn811 4.14.0-115.16.1.el7a.aarch64 #1 SMP Wed Nov 27 13:05:06 EST 2019 aarch64 aarch64 aarch64 GNU/Linux
$ cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.6 (Maipo)
``` 
Git bisect indicates commit 808c80d65ae2e19e79a14a7ecfb67d2788954aaf introduced this problem.  If I revert this commit, the problem vanishes.  We see this problem for other 3.7 versions of python as well.

**Comment by adamjstewart** [comment]
This should be fixed by #14569

**Comment by s-sajid-ali** [comment]
I saw the same error and opened https://github.com/spack/spack/pull/14575 to fix this.

**Comment by paulbry** [comment]
I ran into this today, happy to know there is a fix on the way.

@hppritcha not sure this will work in your case but I was able load a module Python module (`python@3.7.4%gcc@4.8.5` also built with this recent commit) then build `python@3.7.4%gcc@8.3.0`. My system version of Python is still 2 so I had just assumed it was another issue related to that.

I'm using:

```
commit 2eadfa24e961a01b23a54d0c5fb609ae7af4f49a
Author: Todd Gamblin <tgamblin@llnl.gov>
Date:   Tue Jan 21 23:36:10 2020 -0800
```
